mopidy/mpd/protocol/stored_playlists.py

```diff
@@ -145,7 +149,53 @@ def playlistadd(context, name, uri):
 
         ``NAME.m3u`` will be created if it does not exist.
     """
-    raise exceptions.MpdNotImplemented  # TODO
+    uri = context.lookup_playlist_uri_from_name(name)
+    playlist = uri is not None and context.core.playlists.lookup(uri).get()
+    if not playlist:
+        # Create new playlist with this single track
+        lookup_res = context.core.library.lookup(uris=[track_uri]).get()
+        tracks = [a for sl in lookup_res.values() for a in sl]
```

Comment:
The short variables here doesn't immediately map to anything in my mind, so this is a bit hard to follow.